packages/contracts/contracts/ChugSplashManager.sol

```diff
@@ -408,13 +452,44 @@ contract ChugSplashManager is Owned {
         // completed and reset the active bundle hash so that a new bundle can be executed.
         if (bundle.total == bundle.executions.length) {
             bundle.status = ChugSplashBundleStatus.COMPLETED;
+            bytes32 completedBundleId = activeBundleId;
             activeBundleId = bytes32(0);
 
-            emit ChugSplashBundleCompleted(activeBundleId, msg.sender, bundle.total);
+            emit ChugSplashBundleCompleted(completedBundleId, msg.sender, bundle.total);
             registry.announce("ChugSplashBundleCompleted");
+
+            // Refund the executor's bond
+            _returnExecutorBond(completedBundleId);
         }
     }
 
+    /**
+     * @notice **WARNING**: Cancellation is a potentially dangerous action and should not be
+     *         executed unless in an emergency.
+     *
+     *         Cancels an active ChugSplash bundle. This causes the project owner to forfeit their
+     *         bond to the executor, and also refunds the executor's bond.
+     */
+    function cancelActiveChugSplashBundle() public onlyOwner {
+        require(activeBundleId != bytes32(0), "ChugSplashManager: no bundle is currently active");
+
+        ChugSplashBundleState storage bundle = bundles[activeBundleId];
+
+        bytes32 cancelledBundleId = activeBundleId;
+        activeBundleId = bytes32(0);
+        bundle.status = ChugSplashBundleStatus.CANCELLED;
+
+        emit ChugSplashBundleCancelled(cancelledBundleId, msg.sender, bundle.total);
+        registry.announce("ChugSplashBundleCancelled");
+
+        _returnExecutorBond(cancelledBundleId);
+
+        // Send the project owner's bond to the executor
+        address executor = getSelectedExecutor(activeBundleId);
+        (bool success, ) = payable(executor).call{ value: ownerBondAmount }(new bytes(0));
```

Comment:
An interesting issue here is that the executor could prevent a user from cancelling an active bundle if the executor is a contract address that reverts if called by the `Manager` contract. We'll have to use a pull-based model here instead of push-based.